### PR TITLE
Continuous Deployment Flow

### PR DESCRIPTION
**Pull request by hx-markterry**
# What does this change?

Started a doc for how we want our CD to look like.
# Why?

We would like to implement continuous deployment on many of our projects.

### CODE REVIEW

File: cd-flow.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Continuous Deployment Flow
+
+The majority of the automated steps documented below will be handled by our CI setup detail in the [CI Best Practices](ci-best-practices.md).
+
+The process below details what process we are aiming for across the group.
+
+## Planning
+
+1. Discussion with product owner regarding requirements.
```
Comment (kevinhodges) [inline cd-flow.md]: can we get a change to `product owner/stakeholder` please

File: cd-flow.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+1. Work is reviewed by a senior or project guru.
+
+## Silo Testing
+
+1. Tester checks out feature branch on their machine.
+1. Appropriate testing is performed locally.
+
+## Staging (optional)
+
+1. Tester merges feature branch into the `staging` branch.
```
Comment (kevinhodges) [inline cd-flow.md]: `Tester/QA`

File: cd-flow.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# Continuous Deployment Flow
+
+The majority of the automated steps documented below will be handled by our CI setup detail in the [CI Best Practices](ci-best-practices.md).
+
+The process below details what process we are aiming for across the group.
+
+## Planning
+
+1. Discussion with product owner regarding requirements.
+1. Discussion with tester regarding how to test the requirements.
```
Comment (kevinhodges) [inline cd-flow.md]: `tester/QA`

File: cd-flow.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+## Staging (optional)
+
+1. Tester merges feature branch into the `staging` branch.
+1. Changes to staging are automatically deployed to staging environment.
+1. Appropriate testing is performed alongside any other recent changes from the team.
+
+## Go live
+
+1. Tester merges staging branch into the `master` branch.
```
Comment (kevinhodges) [inline cd-flow.md]: `Tester/QA`

**Comment by kevinhodges** [comment]
Looks spot on to me on the whole, excellent stuff.  My only changes are around dialect to cater for SB equivalents.

**Comment by michaelcarter** [comment]
I'm not clear on the separation of roles for the "staging" and "master" branch in this model. It seems similar to "edge" and "master" that we used to have?

**Comment by hx-markterry** [comment]
@michaelcarter correct, its basically the same, I thought that by naming the branch the same as the environment that code deploys to makes things simpler.

**Pull request merged**